staging/src/k8s.io/apiserver/pkg/storage/testing/watcher_tests.go

```diff
@@ -248,18 +277,33 @@ func RunTestWatchError(ctx context.Context, t *testing.T, store InterfaceWithPre
 		t.Fatalf("GuaranteedUpdate failed: %v", err)
 	}
 
-	// Now trigger watch error by injecting failing transformer.
-	revertTransformer := store.UpdatePrefixTransformer(
-		func(previousTransformer *PrefixTransformer) value.Transformer {
-			return &failingTransformer{}
-		})
-	defer revertTransformer()
+	// If we are testing against etcd, we need to deliberately trigger a watch error.
+	// This is not needed while testing with cacher since cacher will emit a resource
+	// expired error itself.
+	if !cacher {
```

Comment:
BTW - can we somehow avoid it?

Given that we create the underlying etcd3 storage for cacher, it should be possible to somehow expose it so that we can actually really update its transformer.
It may require adding some new implementation in etcd3, like:
```
type TestStoreWithUpdatableTransformer struct {
  *store
}

func (...) UpdateTransformer(...) {
  ...
}
```

Requires playing a bit with it, but it should be doable.